apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/DLQDaoIT.java

```diff
@@ -37,7 +37,8 @@ public class DLQDaoIT {
   /** Expiration age in days for deleting expired records. */
   private final int maxAgeDays = 100;
   /** Precomputed value for threshold of record expiration. */
-  private final Instant oldestUpdateDateToKeep = clock.instant().minus(maxAgeDays, ChronoUnit.DAYS);
+  private final Instant oldestUpdateDateToKeep =
+      clock.instant().truncatedTo(ChronoUnit.MICROS).minus(maxAgeDays, ChronoUnit.DAYS);
```

Comment:
As of Java 17, Instant now offers nanosecond precision, however if it uses that precision is based on the machine it running on. When running this locally on my mac, this test passed fine; however running in github actions was returning a diff between the two because of what I'm assuming are machine differences in using clock.instant.

If the target machine does not use nanos, it just truncates the nanoseconds, but this creates a problem when trying to compare timestamps. To remedy this I just truncated it preemptively in the test so it's the same across machines.

See: https://stackoverflow.com/questions/74781495/changes-in-instant-now-between-java-11-and-java-17-in-aws-ubuntu-standard-6-0

Reply:
Nice catch!